src/utils.ts

```diff
@@ -125,23 +126,33 @@ export function getRequiredGazeboDistributions(): string[] {
  *
  * @param requiredGazeboDistributionsList
  * @param ubuntuCodename
+ * @returns Promise<void>
  */
-export function checkUbuntuCompatibility(
+export async function checkUbuntuCompatibility(
 	requiredGazeboDistributionsList: string[],
 	ubuntuCodename: string,
-) {
-	requiredGazeboDistributionsList.forEach((element) => {
-		const compatibleDistros = validGazeboDistroList.find(
-			(obj) => obj.name === element,
-		)!.compatibleUbuntuDistros;
-		if (compatibleDistros.indexOf(ubuntuCodename) <= -1) {
-			throw new Error(
-				"Incompatible Gazebo and Ubuntu combination. \
-        All compatible combinations can be found at \
-        https://gazebosim.org/docs/latest/getstarted/#step-1-install",
-			);
-		}
-	});
+): Promise<void> {
+	await fetch(
+		"https://raw.githubusercontent.com/gazebo-tooling/release-tools/master/jenkins-scripts/dsl/gz-collections.yaml",
+	)
+		.then((response) => response.blob())
+		.then((blob) => blob.text())
+		.then((yamlStr) => {
+			const collections = parseDocument(yamlStr).toJSON();
+			collections["collections"].forEach((obj: any) => {
+				requiredGazeboDistributionsList.forEach((element) => {
+					if (obj["name"] === element) {
+						if (obj["packaging"]["configs"].indexOf(ubuntuCodename) <= -1) {
```

Comment:
We need to be careful here. The packaging.configs is a configuration name that most of the times matches the Ubuntu distro name but NOT always. See https://github.com/gazebo-tooling/release-tools/blob/master/jenkins-scripts/dsl/gz-collections.yaml#L346 noble_no_pr . With that name we need to check in pakaging_configs section in the yaml the system.so.distribution == ubuntu and system.so.version == ubuntuCodename.